components/form/autocompleted/src/index.js

```diff
@@ -143,15 +148,28 @@ export default class FormAutocompleted extends Component {
           {...this.props}
           handleSelect={this._handleSelect}
           active={active}
+          ref={node => { this.suggestList = node }}
           />
         )
       : null
   }
+  _handleOutsideClick = (event) =>
+    !this.excludeFromOutsideClick.includes(event.target) &&
+    this.setState({
+      showSuggestsList: false
+    })
 
   componentDidMount () {
     if (this.state.focus) {
       this.focusInput()
     }
+    this.excludeFromOutsideClick = [
```

Comment:
@JoanClaret pointed out that the tree can be heavy and provoke too many traversing.
This PR should be merged but could be interesting to know if one of you guys knows if it's really an issue.